bigquery/cloud-client/src/main/java/com/example/bigquery/QueryParametersSample.java

```diff
@@ -0,0 +1,275 @@
+/*
+  Copyright 2016 Google Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package com.example.bigquery;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.cloud.bigquery.QueryRequest;
+import com.google.cloud.bigquery.QueryResponse;
+import com.google.cloud.bigquery.QueryResult;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * A sample that demonstrates use of query parameters.
+ */
+public class QueryParametersSample {
+  private static final int ERROR_CODE = 1;
+
+  private static void printUsage() {
+    System.err.println("Usage:");
+    System.err.printf(
+        "mvn exec:java -Dexec.mainClass=%s -Dexec.args=%s\n",
+        QueryParametersSample.class.getCanonicalName(),
+        "sample");
+    System.err.println("sample values: named|array|struct|timestamp");
+    System.err.println("Usage for sample=named:");
+    System.err.println("\tnamed corpus minWordCount");
+    System.err.println("Usage for sample=array:");
+    System.err.println("\tarray gender states...");
+    System.err.println("\tgender=M|F");
+    System.err.println("\tstates=Upper-case 2-letter code for U.S. state, e.g. CA.");
```

Comment:
This help text is a bit hard to understand (hard to tell what's a literal value, and what's a placeholder)... Maybe something like:

```sh
Usage:
    mvn exec:java -Dexec.mainClass=QueryParametersSample -Dexec.args="<sample>"

<sample> can be one of: named, array, struct, timestamp

Usage for <sample>=named:
    mvn exec:java -Dexec.mainClass=QueryParametersSample -Dexec.args="named <corpus> <minWordCount>"

Usage for <sample>=array:
    mvn exec:java -Dexec.mainClass=QueryParametersSample -Dexec.args="array <gender> <states...>"

    where <gender> can be "M" or "F"
    and <states> is any upper-case 2-letter code for a U.S. state, e.g. WA

        mvn exec:java -Dexec.mainClass=QueryParametersSample -Dexec.args="array F MD WA"
```